lib/Monads/trace/Trace_Monad.thy

```diff
@@ -98,9 +98,8 @@ definition bind ::
     | Result (rv, s) \<Rightarrow> fst_upd (\<lambda>ys. ys @ xs) ` g rv s"
 
 text \<open>Sometimes it is convenient to write @{text bind} in reverse order.\<close>
-abbreviation(input) bind_rev ::
-  "('c \<Rightarrow> ('a, 'b) tmonad) \<Rightarrow> ('a, 'c) tmonad \<Rightarrow> ('a, 'b) tmonad" (infixl "=<<" 60)
-  where
+abbreviation (input) bind_rev ::
+  "('c \<Rightarrow> ('a, 'b) tmonad) \<Rightarrow> ('a, 'c) tmonad \<Rightarrow> ('a, 'b) tmonad" (infixl "=<<" 60) where
```

Comment:
@lsf will have to weigh in here, but unless I'm mistaken, according to the current style guide I think once you wrap after the `::` you also are committed to `where` on a separate line.

Reply:
(there are multiple occurrences in other files in this PR as well)

Reply:
I don't have a strong personal opinion on that one, but Raf is right, that is what the style guide doc says.

Reply:
Huh, I missed that last time I looked at the style guide and in an attempt to be consistent have added a lot more like this in the last few PRs.
I'll go through `Trace_Monad` and `Nondet_Monad` and fix all of the definitions to follow this.

Reply:
Where are we between that and changing the style guide? For now probably we should stick with it, but if it goes into AFP who knows what they want anyway. Normally I'm all for conserving vertical space, but in this case I'm not sure.

Reply:
The AFP doesn't care about style, that would be way too hard to enforce. I think it's probably better to keep the rule simple, even if it's not always optimal.